### PR TITLE
Make navbar a fragment

### LINKED ISSUE #89: Make navigation a fragment
**Is this a BUG REPORT or FEATURE REQUEST?**:
feature

**What happened**:
Currently the navigation is "hardcoded" and hidden, when empty.

**What you expected to happen**:
Migrate navigation to its own fragment.

**Anything else we need to know?**:
Navigation fragment would enable stacked navigation and per page navigation.

Needs a bit of thinking.
How to use with auto hide feature?
Where should the menu config be set?
How does it work within the current navigation config from Hugo?
Set menus in config.toml, but set used menu in each fragment?
Too much complexity for multi page sites, that just need one navbar? <-- workaround with a global navbar fragment?

### PR DESCRIPTION
**Pull request by mpourismaiel**
**What this PR does / why we need it**:
Makes navbar a fragment instead of a hardcoded partial.

**Which issue this PR fixes**:
fixes #89 

**Special notes for your reviewer**:
This is based on #118 so please wait for that to be merged.

**Release note**:
```release-note
Navbar is now a fragment
```

### CODE REVIEW

File: layouts/partials/fragments/nav.html [hunk 1]
```diff
@@ -1,3 +1,7 @@
+{{ $nav := .self.Params }}
+{{ "<!-- Nav -->" | safeHTML }}
+{{ $bg := $nav.background | default "secondary" }}
+
 <nav class="overlay navbar navbar-expand-lg py-2
   {{- $bg := .Site.Params.style.background_navigation | default "dark" -}}
```
Comment (stp-ip) [inline layouts/partials/fragments/nav.html]: Seems left over.

**Comment by mpourismaiel** [comment]
@stp-ip Navbar menus are now alongside other menus in `config.toml`. I don't think Hugo would do anything regarding menus in resources and there is now way around it. If I figure out a way to do something like this (this is done in JS, don't now the equivalent in Go or Go templates) it might be less of a pain.

```javascript
.Site.Menus[`navbar-${.self.Params.menu_name}-prepend`]
.Site.Menus[`navbar-${.self.Params.menu_name}-main`]
.Site.Menus[`navbar-${.self.Params.menu_name}-postpend`]
```

File: exampleSite/config.toml [hunk 2]
```diff
@@ -44,6 +44,34 @@ googleAnalytics = ""
   [params.fontawesome]
     disabled = false
 
+# main/prepend/postpend menus are displayed in nav fragment.
+# Keep an eye out for updates. These will be moved out to make navbar more
```
Comment (stp-ip) [inline exampleSite/config.toml]: Great adding a future compatibility change notice here.

**Comment by stp-ip** [comment]
I think we can merge this for now and I'll try to find a Go templates way for the above idea.

**Pull request merged**